lib/nats.js

```diff
@@ -824,10 +843,13 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
   } else {
     psub = 'PUB ' + subject + SPC + opt_reply + SPC;
   }
+
   this.sendCommand(psub + Buffer.byteLength(msg) + CR_LF + msg + CR_LF);
 
   if (opt_callback !== undefined) {
     this.flush(opt_callback);
+  } else if (this.closed) {
+    this.emit('error', PUB_CONN_CLOSED_ERR);
```

Comment:
Also, callbacks are not a substitution for connection-oriented events and their monitoring.  That scope, however, is not generally within the interest of the the piece of code executing this request.

A typical pattern:
```node
var c = new Client();
c.on('error',function(err) {
  console.log("Cry havok!",err);
}

setTimer(function({
  c.publish("hello",function(err) {
    console.log("Oops; I failed to send:",err);
  });
},1000);
```

Completely separate interests for connection events versus request events.  In a real program, you would have entire structures of executions for each of these.  Listening on `error` from the viewpoint of the timer function would be useless.